### DISCUSSION TITLE
Set DebugInfo for help with debugging

### DISCUSSION THREAD

**Discussion by knapsea** [nasa/fprime#918] [Q&A]
For Ref application, in its folder's CMakelist .txt, I  set(CMAKE_BUILD_TYPE Debug) to help step through code and debug.
This doesn't set Debug for the other components and Libraries used by Ref , is there a way to set  debuginfo/"-g" for all components ?
Is there one file this can be enabled at compile time ?

**Comment by Joshua-Anderson** [comment]
Right now the easiest way to do this is by manually setting the debug flag with `set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g")`. Adding proper cmake debug build support to fprime-util is on our todo list. If you're familiar with cmake you could also manually use `cmake -DCMAKE_BUILD_TYPE=Debug ...other flags` to create a debug build.

**Comment by LeStarch** [comment]
I tend to do the following to debug:

```
fprime-util generate --ut <platform or native>
cd fprime-build-automatic-<platform>-ut
make <deployment>_exe
gdb ./bin/*/<deployment>
```

We clearly should make it easier to set debug flags for F´ builds.
